ompi/mpi/c/reduce_local.c

```diff
@@ -76,8 +79,12 @@ int MPI_Reduce_local(const void *inbuf, void *inoutbuf, int count,
     /* Invoke the op component to perform the back-end operation */
     OBJ_RETAIN(op);
     /* XXX -- CONST -- do not cast away const -- update mca/coll */
-    ompi_op_reduce(op,(void *) inbuf, inoutbuf, count, datatype);
+    // Since there is no 'comm' parameter to this interface use 'self' since
+    // this is a local operation to this process.
+    ompi_communicator_t *comm = &ompi_mpi_comm_self.comm;
+    err = comm->c_coll.coll_reduce_local(inbuf, inoutbuf, count, datatype, op,
+                                         comm->c_coll.coll_reduce_local_module);
     OBJ_RELEASE(op);
 
-    return MPI_SUCCESS;
+    OMPI_ERRHANDLER_RETURN(err, MPI_COMM_WORLD, err, FUNC_NAME);
```

Comment:
The use of `MPI_COMM_WORLD` matches the other error handling in the param check earlier in the file.

I used comm_self as the reference point for the `reduce_local`.

Other collectives don't do an `OBJ_RETAIN` / `OBJ_RELEASE` on the datatype so I did not do that here.